Test/Unit/Controller/Cart/DataTest.php

```diff
@@ -0,0 +1,320 @@
+use phpDocumentor\Reflection\Types\Void_;
+use PHPUnit\Framework\TestCase;
+use PHPUnit_Framework_MockObject_MockObject as MockObject;
+
+class DataTest extends TestCase
+{
+    const HINT = 'hint!';
+    const ORDER_REFERENCE = 1234;
+    const TOKEN = 'token';
+    const EXCEPTION_MESSAGE = 'Test exception message';
+
+    /**
+     * @var Context
+     */
+    private $context;
+
+    /**
+     * @var MockObject|Cart
+     */
+
+    private $cartHelper;
+
+    /**
+     * @var MockObject|Config
+     */
+
+    private $configHelper;
+
+    /**
+     * @var MockObject|Bugsnag
+     */
+
+    private $bugsnag;
+
+    /**
+     * @var RequestInterface request
+     */
+    private $request;
+
+    /**
+     * @var MetricsClient metricsClient
+     */
+    private $metricsClient;
+
+    private $response;
+
+    private $responseData;
+
+    private $expectedCart;
+
+    /**
+     * @var Object requestShippingAddress
+     */
+    private $requestShippingAddress;
+
+    /**
+     * @var MockObject | Response boltpayOrder
+     */
+    private $boltpayOrder;
+
+    protected function setUp()
+    {
+        $this->initResponseData();
+        $this->initRequiredMocks();
+    }
+
+    private function initResponseData()
+    {
+        $this->requestShippingAddress = 'String';
+
+        $this->response = (object) ( array(
+            'cart' =>
+                (object) ( array(
+                    'order_reference' => self::ORDER_REFERENCE,
+                    'display_id'      => '100050001 / 1234',
+                    'shipments'       =>
+                        array(
+                            0 =>
+                                (object) ( array(
+                                    'shipping_addres' => $this->requestShippingAddress,
+                                    'shipping_method' => 'unknown',
+                                    'service'         => 'Flat Rate - Fixed',
+                                    'cost'            =>
+                                        (object) ( array(
+                                            'amount'          => 500,
+                                            'currency'        => 'USD',
+                                            'currency_symbol' => '$',
+                                        ) ),
+                                    'tax_amount'      =>
+                                        (object) ( array(
+                                            'amount'          => 0,
+                                            'currency'        => 'USD',
+                                            'currency_symbol' => '$',
+                                        ) ),
+                                    'reference'       => 'flatrate_flatrate'
+                                ) ),
+                        ),
+                ) ),
+            'token' => self::TOKEN
+        ) );
+
+        $this->responseData = json_decode(json_encode($this->response), true);
```

Comment:
I know this was copied from the actual code, but a short comment explaining why this is done would be good.